sympy/polys/powerseriesring.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations
+
+from typing import Any, Protocol, TypeAlias, runtime_checkable
+
+from sympy.external.gmpy import GROUND_TYPES
+from sympy.polys.domains import Domain
+from sympy.utilities import public
+
+
+ZZSeries: TypeAlias = Any
+QQSeries: TypeAlias = Any
```

Comment:
It would be better to have a TypeVar here:
```
Tseries = TypeVar('Tseries')
```
Then the procols can be generic over `Tseries` and the QQ protocol can inherit the ZZ protocol without changing any methods.